src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs

```diff
@@ -1,83 +1,514 @@
+            var request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(string.Format("{0}/subscriptions?api-version=2014-04-01-preview", baseuri))
+            };
+
+            HttpClient client = new HttpClient();
+
+            credentials.ProcessHttpRequestAsync(request, CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+
+            HttpResponseMessage response = client.SendAsync(request).Result;
+            response.EnsureSuccessStatusCode();
+
+            string jsonString = response.Content.ReadAsStringAsync().Result;
+
+            var jsonResult = JObject.Parse(jsonString);
+            var results = ((JArray)jsonResult["value"]).Select(item => new SubscriptionInfo((JObject)item)).ToList();
+            return results;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="testEnv"></param>
+        /// <param name="connectionString"></param>
+        private void SetEnvironmentSubscriptionId()
+        {
+            if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                if (HttpMockServer.Variables.ContainsCaseInsensitiveKey(ConnectionStringKeys.SubscriptionIdKey))
+                {
+                    this.SubscriptionId = HttpMockServer.Variables.GetValueUsingCaseInsensitiveKey(ConnectionStringKeys.SubscriptionIdKey);
+                }
+                else if (string.IsNullOrEmpty(this.SubscriptionId))
+                {
+                    throw new Exception(string.Format("Subscription Id is not present in the recorded mock or in connection string (e.g. {0}=<subscriptionId>)", ConnectionStringKeys.SubscriptionIdKey));
+                }
+            }
+            else // HttpRecorder Mode == Record/None
+            {
+                // Preserve/restore subscription ID
+                HttpMockServer.Variables.UpdateDictionary(ConnectionStringKeys.SubscriptionIdKey, this.SubscriptionId);
+            }
+        }
+
+        /// <summary>
+        /// TODO: not sure if we need this for the ease to get the value of various connection string keys
+        /// Currently you need to use the below syntax to get to the value of each connectionstring keys
+        /// this.ConnectionString.KeyValuePairs[ConnectionStringKeys.SubscriptionIdKey]
+        /// with this funciton it will be as below
+        /// this.GetKeyValue(ConnectionStringKeys.SubscriptionIdKey)
+        /// </summary>
+        /// <param name="connectionStringKeyName"></param>
+        /// <returns></returns>
+        private string GetKeyValue(string connectionStringKeyName)
+        {
+            string keyValue = string.Empty;
+            if (this.ConnectionString.KeyValuePairs.ContainsKey(connectionStringKeyName))
+            {
+                this.ConnectionString.KeyValuePairs.TryGetValue(connectionStringKeyName, out keyValue);
+            }
+
+            return keyValue;
+        }
+
+        #endregion
+    }
+
+/*
+
```

Comment:
please delete all these commented code. it is hard to understand which code is commented out and which one is not.